lib/exiffer/rewrite.ex

```diff
@@ -10,27 +10,50 @@ defmodule Exiffer.Rewrite do
   alias Exiffer.JPEG.{Entry, IFD, IFDBlock}
   alias Exiffer.JPEG.Header.APP1.EXIF
 
-  def set_date_time(source, destination, %DateTime{} = date_time) do
-    set_date_time(source, destination, DateTime.to_naive(date_time))
-  end
-
-  def set_date_time(source, destination, %NaiveDateTime{} = date_time) do
-    Logger.info("Exiffer.Rewrite.set_date_time/3")
+  def rewrite(source, destination, rewrite_fun) when is_function(rewrite_fun, 1) do
     input = Buffer.new(source)
     {jpeg, input} = Exiffer.parse(input)
-    headers = internal_set_date_time(jpeg.headers, date_time)
 
-    Logger.debug "Setting initial byte order to :big"
-    Binary.set_byte_order(:big)
+    headers = rewrite_fun.(jpeg.headers)
 
+    Binary.set_byte_order(:big)
     output = Buffer.new(destination, direction: :write)
     Buffer.write(output, JPEG.magic())
-    :ok = Exiffer.Serialize.write(headers, output.io_device)
 
+    :ok = Exiffer.Serialize.write(headers, output.io_device)
     :ok = Buffer.close(input)
     :ok = Buffer.close(output)
 
-    {:ok}
+    :ok
+  end
+
+  def set_make_and_model(source, destination, make, model) do
```

Comment:
It would be good to have a test for this

Reply:
OK, already add test for this on [here](https://github.com/joeyates/exiffer/pull/1/commits/4d67c31ec94a62e5d323fbda7883c7e636887f89)